[tensorexpr] aten::Linear external call

IvanKobzarev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #64751

Differential Revision: [D30842102](https://our.internmc.facebook.com/intern/diff/D30842102)

IvanKobzarev committed:
[tensorexpr] Handle linear with addmm

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30842102).

IvanKobzarev committed:
Update on "[tensorexpr] Handle linear with addmm"


Differential Revision: [D30842102](https://our.internmc.facebook.com/intern/diff/D30842102)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30842102).

IvanKobzarev committed:
Update on "[tensorexpr] Handle linear with addmm"


Differential Revision: [D30842102](https://our.internmc.facebook.com/intern/diff/D30842102)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30842102).

IvanKobzarev committed:
Update on "[tensorexpr] Handle linear with addmm"


Differential Revision: [D30842102](https://our.internmc.facebook.com/intern/diff/D30842102)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30842102).

navahgar commented on torch/csrc/jit/tensorexpr/kernel.cpp:
```diff
@@ -2523,6 +2523,20 @@ Tensor tensorexpr::computeOperandValue(
+      addmmInputs.push_back(inputs[0]);
+      addmmInputs.push_back(inputs[1]);
+      // Using int64_t as computeAddMM supports only int64_t scalars now
+      addmmInputs.push_back(1l); // beta
```
s/1l/1ll

navahgar commented on torch/csrc/jit/tensorexpr/kernel.cpp:
```diff
@@ -2523,6 +2523,20 @@ Tensor tensorexpr::computeOperandValue(
+      addmmInputs.push_back(inputs[1]);
+      // Using int64_t as computeAddMM supports only int64_t scalars now
+      addmmInputs.push_back(1l); // beta
+      addmmInputs.push_back(1l); // alpha
```
s/1l/1ll

navahgar reviewed (commented):
What about a test for this?

IvanKobzarev committed:
Update on "[tensorexpr] Handle linear with addmm"


Differential Revision: [D30842102](https://our.internmc.facebook.com/intern/diff/D30842102)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30842102).

IvanKobzarev committed:
Update on "[tensorexpr] Handle linear with addmm"


Differential Revision: [D30842102](https://our.internmc.facebook.com/intern/diff/D30842102)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30842102).

IvanKobzarev committed:
Update on "[tensorexpr] Handle linear with addmm"


Differential Revision: [D30842102](https://our.internmc.facebook.com/intern/diff/D30842102)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30842102).

IvanKobzarev committed:
Update on "[tensorexpr] Handle linear with addmm"


Differential Revision: [D30842102](https://our.internmc.facebook.com/intern/diff/D30842102)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30842102).

IvanKobzarev committed:
Update on "[tensorexpr] Handle linear with addmm"


Differential Revision: [D30842102](https://our.internmc.facebook.com/intern/diff/D30842102)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30842102).

IvanKobzarev committed:
Update on "[tensorexpr] Handle linear with addmm"


Differential Revision: [D30842102](https://our.internmc.facebook.com/intern/diff/D30842102)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30842102).

IvanKobzarev committed:
Update on "[tensorexpr] Handle linear with addmm"


Differential Revision: [D30842102](https://our.internmc.facebook.com/intern/diff/D30842102)

[ghstack-poisoned]

navahgar reviewed (approved):
LGTM

IvanKobzarev committed:
Update on "[tensorexpr] aten::Linear external call"


Differential Revision: [D30842102](https://our.internmc.facebook.com/intern/diff/D30842102)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30842102).
